shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -272,7 +272,11 @@ private static int inputTypeFromTextInputType(
       textType |= InputType.TYPE_TEXT_VARIATION_PASSWORD;
     } else {
       if (autocorrect) textType |= InputType.TYPE_TEXT_FLAG_AUTO_CORRECT;
-      if (!enableSuggestions) textType |= InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS;
+      if (!enableSuggestions) {
+        // Note: both required. Some devices ignore TYPE_TEXT_FLAG_NO_SUGGESTIONS.
+        textType |= InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS;
+        textType |= InputType.TYPE_TEXT_VARIATION_PASSWORD;
```

Comment:
Why this flag versus `TYPE_TEXT_VARIATION_VISIBLE_PASSWORD`? The links you mentioned point to that flag, the concern I assume being that this may obscure what the user is typing.

Reply:
You are perfectly right that the flag should be `TYPE_TEXT_VARIATION_VISIBLE_PASSWORD`. While testing on my device (Pixel 7 / GBoard), I did not notice this mistake because the keyboard is the expected one in both cases.

> this may obscure what the user is typing.

This won't have such an effect because the obscure logic is on the Dart side. Nonetheless, this is a really good catch because using the right value makes this PR coherent with the links listed in its description.